spark/v3.3/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -74,6 +74,7 @@ statement
     | ALTER TABLE multipartIdentifier SET IDENTIFIER_KW FIELDS fieldList                    #setIdentifierFields
     | ALTER TABLE multipartIdentifier DROP IDENTIFIER_KW FIELDS fieldList                   #dropIdentifierFields
     | ALTER TABLE multipartIdentifier CREATE BRANCH identifier (AS OF VERSION snapshotId)? (RETAIN snapshotRefRetain snapshotRefRetainTimeUnit)? (snapshotRetentionClause)?    #createBranch
+    | ALTER TABLE multipartIdentifier REPLACE BRANCH identifier (AS OF VERSION snapshotId)? (RETAIN snapshotRefRetain snapshotRefRetainTimeUnit)? (snapshotRetentionClause)?  #replaceBranch
```

Comment:
After writing this out, starting to change my mind :) maybe we combine {CREATE | REPLACE} in the same definition if that's possible?

Reply:
Agree, I think in this case there are a few arguments for combining it:
1. syntax is exactly the same, and adding replace does not complicate the logic too much
2. CREATE/REPLACE table follows a similar pattern
3. At API level, we also have similar thing like `createOrReplaceTableTransaction`